activerecord/lib/active_record/railties/databases.rake

```diff
@@ -95,9 +95,12 @@ db_namespace = namespace :db do
 
   # IMPORTANT: This task won't dump the schema if ActiveRecord::Base.dump_schema_after_migration is set to false
   task :_dump do
-    if ActiveRecord::Base.dump_schema_after_migration
+    db_config = ActiveRecord::DatabaseConfigurations.new(databases).configs_for(env_name: Rails.env, name: "primary")
```

Comment:
This won't work correctly. The `databases` is the initial setup which strips all ERB so the configuration may be invalid. That's why we only use it for generating the name of the tasks.

So to fix this it needs to be updated to do the following:

1) `task :_dump do` needs to be `task :_dump => :load_config do`
2) `db_config = ActiveRecord::DatabaseConfigurations.new(databases).configs_for(env_name: Rails.env, name: "primary")` needs to be `db_config = ActiveRecord::Base.configurations.configs_for(env_name: Rails.env, name: "primary")`